### PR TITLE
Added basic Typescript definitions.

### PR DESCRIPTION
**Pull request by andrewstart**
Added a basic Typescript definition file. I am relatively new to Typescript, so I don't know how to improve it from here.

### CODE REVIEW

**Comment by dantman** [comment]
It's a start, but the interface has changed quite significantly. A proper Typescript definition should include the `Facebook` class.

**Comment by andrewstart** [comment]
Updated to have all of the module exports and have the .api() Promises.

**Comment by dantman** [comment]
I'm still not sure how I should add tests for this to ensure it always works.

But in the meantime here are some notes:
* The `FB.{api,napi}(path, method, cb?)` form is also valid (ie: +`method` but without `params`)
* Missing `extend` and `withAccessToken`
* Options to `getLoginUrl` are technically optional
* Bonus: A HTTPMethods enum could be made for the method argument
* Bonus: You should be able to define a FacebookOptions interface with the list of option names and their type, that can then be applied to the return type of `options()` and to `options(optionsDict: any)`

**Comment by dantman** [comment]
[typescript-definition-tester](https://www.npmjs.com/package/typescript-definition-tester) may work for adding tests for this.

**Comment by andrewstart** [comment]
I've updated the definitions as per your suggestions, and also threw in an interface for `getLoginUrl()`. 

I also added a test that uses typescript-definition-tester. The test uses ES6 import syntax. I tried to set up an additional file to test that uses `require()` as well, but I haven't found a way for a TypeScript module to have both the ES6 and CommonJS exports. As such, TypeScript users should stick to the ES6 imports, because that's what this definition is set up for.

File: fb.d.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import Promise = require("any-promise");
```
Comment (ibratoev) [inline fb.d.ts]: You can remove this line. `Promise` can be used directly without importing it as it is part of ES6.

Comment (andrewstart) [inline fb.d.ts]: While that is true, that doesn't mean that everyone is using an ES6 environment, and additionally this library uses `any-promise` as the Promise implementation.

Comment (ibratoev) [inline fb.d.ts]: That's how i have seen it used in most well written typings I have used. If you are not targeting ES6, it is up to the consumer to provide the typings for `Promise`. Example from a fairy recent declaration https://github.com/facebook/dataloader/blob/master/src/index.d.ts

Comment (ibratoev) [inline fb.d.ts]: Additionally, 2.0 of this library supports only NodeJS environments that support `Promise`-s natively ;)

**Comment by ibratoev** [comment]
Btw, my comment is minor and not a blocker. I am 👍 for merging the typings. Tested it locally - all works fine ;) Thanks for that, @andrewstart ;)

**Comment by dantman** [comment]
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html seems to recommend publishing types through DefinitelyTyped/`@types` instead of as part of the package.

In any case I still want to get flow working first. Maybe we could event convert that to typescript definitions.

**Comment by nilfalse** [comment]
So, no plan to get this merged any time soon?

**Comment by dantman** [comment]
@nilfalse The code in this PR isn't even up to date, the API has changed since this was written.

**Comment by dantman** [comment]
This PR is out of date, i.e. it will not work anymore. And it does not attempt to handle Flow. #167 documents plans regarding both Flow and TypeScript. The code in this PR may be helpful in the future as a reference in for writing up to date typings, but for now I am closing it.

**Comment by user734853459690357** [comment]
pzdc, why this pr wasn't accept?

**Pull request closed without merging**